### PR TITLE
Pelasne/add whitelist tenant

### PR DESCRIPTION
**Pull request by plasne**
Add whitelist tenants.

### CODE REVIEW

File: bot/services/ChatBot.cs [hunk 1]
```diff
@@ -44,6 +44,37 @@ public class ChatBot(
...
+        // compare the tenant
+        try
+        {
+            string tenantId = turnContext.Activity.ChannelData["tenant"]["id"].ToString();
+            if (this.config.VALID_TENANTS.Contains(tenantId))
+            {
+                this.logger.LogDebug("the tenant {t} is authorized to use the bot.", tenantId);
+                return true;
+            }
+            else
```
Comment (seekdavidlee) [inline bot/services/ChatBot.cs]: it may be cleaner to just return false without the else

File: bot/services/ChatBot.cs [hunk 2]
```diff
@@ -334,6 +365,12 @@ await foreach (var chatResponse in streamingCall.ResponseStream.ReadAllAsync(can
 
     protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
     {
+        // verify authorization
+        if (!this.IsAuthorized(turnContext))
+        {
+            throw new Exception("the activity was not authorized to use this bot.");
```
Comment (seekdavidlee) [inline bot/services/ChatBot.cs:369]: suggest to say: the tenant was not authorized to use this bot

Comment (seekdavidlee) [inline bot/services/ChatBot.cs:369]: will this return a 403 later or how does this work?

File: bot/services/ChatBot.cs (hunk 1 above)
Comment (plasne) [reply bot/services/ChatBot.cs]: sure, I can do that.

File: bot/services/ChatBot.cs (hunk 2 above)
Comment (plasne) [reply bot/services/ChatBot.cs:369]: The log does refer to the tenant, but outside the IsAuthorized method we don't know the reason it was denied. In theory there could be other checks than just tenant.

Comment (plasne) [reply bot/services/ChatBot.cs:369]: This is gRPC, so definitely not a 403. This raises to the AdapterWithErrorHandler. How we decide to handle that has been moved outside of the scope of this engagement, but you could imagine they might decide to respond to the user in some different way.

**Review by seekdavidlee — APPROVED** [review]

**Pull request merged**